dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -52,25 +65,61 @@
 class StaticLinkageChecker {
 
   /**
-   * Given list of the jar files as file names in filesystem, outputs the report of static linkage
-   * check.
+   * Given Maven coordinates or list of the jar files as file names in filesystem, outputs the
+   * report of static linkage check.
    *
-   * @param arguments jar files available in the file system
    * @throws IOException when there is a problem in reading a jar file
    * @throws ClassNotFoundException when there is a problem in reading a class from a jar file
+   * @throws RepositoryException when there is a problem in resolving the Maven coordinate to jar
+   *     files
    */
-  public static void main(String[] arguments) throws IOException, ClassNotFoundException {
+  public static void main(String[] arguments)
+      throws IOException, ClassNotFoundException, RepositoryException {
+    Options options = new Options();
```

Comment:
This logic is complex enough to pull into a separate method.

Reply:
+1 I usually like to have option parsing and the program logic in separate functions

Reply:
Fixed